inference-engine/src/mkldnn_plugin/mkldnn_node.cpp

```diff
@@ -1537,6 +1551,69 @@ bool MKLDNNNode::canBePerformedAsScaleShift(const MKLDNNNode *parentNode) const
             || isConvertablePowerStatic();
 }
 
+bool MKLDNNNode::inputShapesDefined() const {
+    for (size_t i = 0; i < getParentEdges().size(); i++) {
+        if (!getParentEdgesAtPort(i)[0]->getMemory().getDesc().isDefined())
+            return false;
+    }
+    return true;
+}
+
+bool MKLDNNNode::needPrepareParams() const {
+    return inputShapesModified();
+}
+
+bool MKLDNNNode::inputShapesModified() const {
+    if (lastInputDims.size() != getParentEdges().size()) {
+        if (lastInputDims.empty())
+            return true;
+        IE_THROW() << "Input dims and parent edges number mismatch!";
+    }
+
+    for (size_t i = 0; i < lastInputDims.size(); i++) {
+        if (lastInputDims[i] != getParentEdgesAtPort(i)[0]->getMemory().getStaticDims())
+            return true;
+    }
+    return false;
+}
+
+bool MKLDNNNode::needShapeInfer() const {
+    return inputShapesModified();
+}
+
+std::vector<VectorDims> MKLDNNNode::shapeInfer() const {
+    for (size_t i = 0; i < opToShapeInfer->get_input_size(); i++) {
+        if (!dynamic_cast<ngraph::opset1::Constant *>(opToShapeInfer->get_input_node_ptr(i))) {
+            opToShapeInfer->get_input_tensor(i).set_partial_shape(
+                getParentEdgesAtPort(i)[0]->getMemory().getDesc().getShape().toPartialShape());
+        }
+    }
+
+    opToShapeInfer->validate_and_infer_types();
+
+    IE_ASSERT(opToShapeInfer->get_output_size() == outputShapes.size());
+
+    std::vector<VectorDims> newOutputShapes(outputShapes.size());
+    for (size_t i = 0; i < newOutputShapes.size(); i++) {
+        const auto &partShape = opToShapeInfer->get_output_partial_shape(i);
+        if (partShape.is_dynamic())
+            IE_THROW(NotImplemented) << "CPU plug-in doesn't support default shape infer for nodes with internal dynamism";
+        newOutputShapes[i] = partShape.get_shape();
+    }
+    return newOutputShapes;
+}
+
+void MKLDNNNode::updateLastInputDims() {
+    if (lastInputDims.size() != getParentEdges().size()) {
+        if (!lastInputDims.empty())
+            IE_THROW() << "Input dims and parent edges number mismatch!";
+        lastInputDims.resize(getParentEdges().size());
+    }
```

Comment:
```suggestion
    if (lastInputDims.empty()) {
        lastInputDims.resize(getParentEdges().size());
    }
    if (lastInputDims.size() != getParentEdges().size()) {
            IE_THROW() << "Input dims and parent edges number mismatch!";
    }
```

Reply:
in this case we will be call `resize` on every inference for input node